### PR TITLE
Add type mapping section to Google Sheets connector

### PR DESCRIPTION
**Pull request by Jessie212**
## Description

Add type mapping section to Google Sheets connector.

## Non-technical explanation

Add a section explaining type mappings between Google Sheets and Trino.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/connector/googlesheets.rst [hunk 1]
```diff
@@ -98,6 +98,86 @@ may prevent the limit from being reached. Running queries on the ``information_s
 table without a schema and table name filter may lead to hitting the limit, as this requires
 fetching the sheet data for every table, unless it is already cached.
 
+Type mapping
+------------
+
+Because Trino and Google Sheets each support types that the other does not, this
+connector :ref:`modifies some types <type-mapping-overview>` when reading data.
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/googlesheets.rst:105]: The connector supports only `TEXT` → `VARCHAR` mapping. Other types will be added in https://github.com/trinodb/trino/pull/5011

File: docs/src/main/sphinx/connector/googlesheets.rst [hunk 2]
```diff
@@ -98,6 +98,86 @@ may prevent the limit from being reached. Running queries on the ``information_s
...
+    - ``DOUBLE``
+    -
+  * - ``ARRAY``
+    - ``ARRAY``
+    -
+  * - ``STRING``
+    - ``VARCHAR``
+    -
+  * - ``TEXT``
+    - ``CHAR``
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/googlesheets.rst:133]: What's the difference between `STRING` and `TEXT`?

File: docs/src/main/sphinx/connector/googlesheets.rst [hunk 3]
```diff
@@ -98,6 +98,86 @@ may prevent the limit from being reached. Running queries on the ``information_s
...
+  * - ``DATE``
+    - ``DECIMAL (p,s)``
+    -
+  * - ``JSON``
+    - ``JSON``
+    -
+
+No other types are supported.
+
+Trino type to Google Sheets type mapping
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/googlesheets.rst:144]: The connector is read-only for now. Let's remove this mapping table. Write operations will be supported in https://github.com/trinodb/trino/pull/5011

File: docs/src/main/sphinx/connector/googlesheets.rst (hunk 2 above)
Comment (Jessie212) [reply docs/src/main/sphinx/connector/googlesheets.rst:133]: This was an additional inclusion taken from an outside source. I removed it.

File: docs/src/main/sphinx/connector/googlesheets.rst (hunk 1 above)
Comment (ebyhr) [reply docs/src/main/sphinx/connector/googlesheets.rst:105]: I meant only one entry should exist at this time.

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**